### PR TITLE
SmartScripts: Implement Pethandling for SMART_ACTION_CALL_KILLEDMONSTER

### PR DESCRIPTION
**Pull request by Arsul**
Fixes the situation that if a Pet does the lasthit, the Owner will still get the Questcredit.

### CODE REVIEW

**Comment by Shauren** [comment]
tabssssssssssss

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -859,45 +859,52 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
...
+					}
+					else if (IsUnit(*itr)) // Special handling for vehicles
+					{
+						if (Vehicle* vehicle = (*itr)->ToUnit()->GetVehicleKit())
+						{
+							for (SeatMap::iterator itr = vehicle->Seats.begin(); itr != vehicle->Seats.end(); ++itr)
+								if (Player* player = ObjectAccessor::FindPlayer(itr->second.Passenger.Guid))
+									player->KilledMonsterCredit(e.action.killedMonster.creature);
+						}
+						else if (Unit* owner = (*itr)->ToPlayer()->GetOwner()) {
```
Comment (joschiwald) [inline src/server/game/AI/SmartScripts/SmartScript.cpp]:
```c++
if (IsUnit(*itr))
    if (Unit* owner = (*itr)->ToPlayer()->GetOwner())
```
will never work

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 2]
```diff
@@ -859,45 +859,52 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
...
+				for (ObjectList::const_iterator itr = targets->begin(); itr != targets->end(); ++itr)
+				{
+					if (IsPlayer(*itr))
+					{
+						(*itr)->ToPlayer()->KilledMonsterCredit(e.action.killedMonster.creature);
+						TC_LOG_DEBUG("scripts.ai", "SmartScript::ProcessAction: SMART_ACTION_CALL_KILLEDMONSTER: Player %u, Killcredit: %u",
+							(*itr)->GetGUIDLow(), e.action.killedMonster.creature);
+					}
+					else if (IsUnit(*itr)) // Special handling for vehicles
+					{
```
Comment (cemak) [inline src/server/game/AI/SmartScripts/SmartScript.cpp]: Maybe:

Unit* owner = (*itr)->ToUnit()->GetOwner())
if (owner && owner->GetTypeId() == TYPEID_PLAYER)
{
    if (Player* player = owner->ToPlayer())
    {
       player->KilledMonsterCredit(e.action.killedMonster.creature);
       TC_LOG_DEBUG("scripts.ai", "SmartScript::ProcessAction: SMART_ACTION_CALL_KILLEDMONSTER: Player %u, Killcredit: %u", player->GetGUIDLow(), e.action.killedMonster.creature)
    }
}

File: src/server/game/AI/SmartScripts/SmartScript.cpp (hunk 1 above)
Comment (cemak) [inline src/server/game/AI/SmartScripts/SmartScript.cpp]: Maybe:
http://pastebin.com/eWB0pq6w

**Comment by DDuarte** [comment]
TODO: Port to 6.x. (PR was closed automatically by a branch deletion.)

**Comment by Aokromes** [comment]
more than 1 year and still rooting.

**Comment by tkrokli** [comment]
Is it OK if I create a new PR for this content, add the suggestion from **cemak** and then just let any dev suggest what kind of changes are needed? (Looks like the original author has not been active since this PR was opened 1.5 years ago.)

**Comment by Aokromes** [comment]
Yes @tkrokli

**Comment by Aokromes** [comment]
```code
[13:21:14] <Treeston> should use SMART_TARGET_LOOT_RECIPIENTS instead
[13:21:17] <Treeston> using killing blow is a hack
[13:21:27] <Treeston> ignores creature tagging

**Pull request closed without merging**